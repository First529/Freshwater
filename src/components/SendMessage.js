import React from "react";

export default class SendMessage extends React.Component {
  constructor(props) {
    super(props);
    this.state = { message: "" };
    this.handleChange = this.handleChange.bind(this);
    this.handleKeyPress = this.handleKeyPress.bind(this);
  }

  handleChange(event) {
    this.setState({ message: event.target.value });
  }

  handleKeyPress = event => {
    var text = this.state.message.replace(/\s/g, "");
    if (event.key === "Enter" && text !== "") {
      this.props.sendMessage(this.state.message);
      this.setState({ message: "" });
    }
  };

  render() {
    return (
      <div className="send-message">
        <input
          placeholder="Write your message here"
          value={this.state.message}
          type="message"
          onChange={this.handleChange}
          onKeyPress={this.handleKeyPress}
        />
      </div>
    );
  }
}
