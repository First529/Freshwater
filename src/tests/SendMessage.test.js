import React from "react";
import { shallow } from "enzyme";
import SendMessage from "../components/SendMessage";

describe("SendMessage", () => {
  it("renders correctly", () => {
    const component = shallow(<SendMessage />);
    expect(component).toMatchSnapshot();
  });

  it("check if message is empty", () => {
    const component = shallow(<SendMessage />);
    expect(component.state("message")).toBe("");
  });
});
