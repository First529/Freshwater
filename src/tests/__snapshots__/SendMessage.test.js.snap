// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SendMessage renders correctly 1`] = `
<div
  className="send-message"
>
  <input
    onChange={[Function]}
    onKeyPress={[Function]}
    placeholder="Write your message here"
    type="message"
    value=""
  />
</div>
`;
